extra_foam/gui/image_tool/reference_view.py

```diff
@@ -99,76 +103,87 @@ def _loadReference(self):
         if filepath:
             self._pub.set(filepath)
             self._ctrl_widget.filepath_le.setText(filepath)
+            self._ctrl_widget.save_btn.setDisabled(True)
 
     @pyqtSlot()
-    def _setReference(self):
+    def _setCurrentAsReference(self):
         """Set the current corrected image as reference."""
-        img = self._corrected.image
-        if img is not None:
-            filepath = osp.join(ROOT_PATH, "tmp", ".reference.npy")
-            if not osp.exists(osp.dirname(filepath)):
-                os.mkdir(osp.dirname(filepath))
+        image = self._corrected.image
+        if image is not None:
+            self._write_temp_reference_file(image)
+            if self._check_reference_file():
+                self._set_reference(REFERENCE_FILE)
+
+    def _write_temp_reference_file(self, image):
+        if image is None:
+            return
 
-            try:
-                write_image(filepath, img)
-            except ValueError as e:
-                logger.error(str(e))
+        if not osp.exists(osp.dirname(REFERENCE_FILE)):
+            os.mkdir(osp.dirname(REFERENCE_FILE))
 
-            self._pub.set(filepath)
-            self._ctrl_widget.filepath_le.setText(filepath)
+        try:
+            write_image(REFERENCE_FILE, image)
+        except ValueError as e:
+            logger.error(str(e))
+            # Remove previous reference file
+            os.remove(REFERENCE_FILE)
+
+    def _set_reference(self, path):
+        """Sets the input image as reference"""
+        self._pub.set(path)
+        self._ctrl_widget.filepath_le.setText(path)
 
     @pyqtSlot()
     def _removeReference(self):
         """Remove the reference image."""
-        self._pub.set("")
-        self._ctrl_widget.filepath_le.setText("")
+        self._set_reference("")
+        self._reference.clear()
 
     @pyqtSlot(bool)
     def _recordReference(self, is_recording):
+        """Records the reference image aby the"""
```

Comment:
I don't get this comment (typo? incomplete?) ... (maybe not of great significance anyway)